libs/modelviewer/src/hooks/useSelectionService.ts

```diff
@@ -0,0 +1,32 @@
+import { CogniteCadModel } from '@cognite/reveal';
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import { SelectionService } from '../services';
+import { useModelContext, useModelViewerContext } from '../providers';
+
+export const useSelectionService = () => {
+  const { echoInstance } = useModelViewerContext();
+  const { modelMeta } = useModelContext();
+
+  const [model, setModel] = useState<CogniteCadModel>();
+
+  const viewer = useMemo(() => echoInstance?.viewer, [echoInstance]);
+
+  const hierarchyClient = useMemo(() => echoInstance?.hierarchyApiClient, [echoInstance]);
+
+  const selectionService = useMemo(() => {
+    if (model && hierarchyClient && modelMeta && viewer)
+      return new SelectionService(modelMeta, hierarchyClient, viewer, model);
+  }, [modelMeta, hierarchyClient, viewer, model]);
+
+  const loadModel = useCallback(async () => {
+    if (!viewer || !modelMeta) return;
+    const loadedModel = await viewer.loadModel(modelMeta);
+    setModel(loadedModel);
+  }, [viewer, modelMeta]);
+
+  useEffect(() => {
+    loadModel();
```

Comment:
Could we use react query or something for this. Working with asynchronous code in a useffect is not the best. Using react query would also remove the need for all the callbacks and memoization

Reply:
Fixed